File: C/exercice3_5.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>
#include <time.h>
#include <errno.h>

int main()
{
	char* texte = malloc(50) ;
	pid_t fils ;
	while (1)
	{
		printf("Saisir un nom de repertoire:(taper quitter pour arrêter): ");
		scanf("%s", texte) ;
		if (strncmp(texte, "quitter", 7) == 0)
		{
			return (0) ;
		}
		fils = fork() ;
		if (fils == 0)
		{
			execlp("vim", "vim", texte, NULL) ;
		}
		else
		{
			wait(NULL) ;
		}
	}
	return 0;
}

File: Python/exercice3_5.py
import os

def main():
    pwd = os.getenv("PWD")
    
    while True:
        texte = input("Saisir un nom de fichier (taper 'quitter' pour arrêter): ")
        
        if repertoire == "quitter":
            return 0

        fils = os.fork()

        if fils == 0:
            os.execlp("vim", "vim", "-r", texte, None)

        elif fils > 0:
            os.wait()
    
if __name__ == "__main__":
    main()